### DISCUSSION TITLE
S3 bucket to keep only 3 new objects

### DISCUSSION THREAD

**Discussion by itsyosefali** [aws/aws-cli#7687] [Q&A] [answered]
### Describe the feature

as you see from the title , how can i make my amazon s3 bucket to only keep the newest 3 objects (they don't have the same name so i didn't enable the version) and delete the old ones ?

Ps: any help without using lambda function cus i read that it takes money per request

### Use Case

lifecycle

### Acknowledgements

- [x] I may be able to implement this feature request
- [x] This feature might incur a breaking change

### CLI version used

-

### Environment details (OS name and version, etc.)

-

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @itsyosefali, thanks for reaching out.  I'm going to convert this into a discussion, since it is more of a guidance question that is better suited as a discussion.  

As mentioned in the [bucket limitations documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html), buckets have no limit on the number of objects they can contain.  If your objects are to be uploaded on a regular schedule, you could use [bucket lifecycles](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html) to have older objects be deleted.  Otherwise, you're right that you'd have to use Lambda or some other service to automate these deletions.  Please let me know if you have any other questions.

**Reply by itsyosefali** [reply]
> Hi @itsyosefali, thanks for reaching out.  I'm going to convert this into a discussion, since it is more of a guidance question that is better suited as a discussion.  
> 
> As mentioned in the [bucket limitations documentation](https://docs.aws.amazon.com/AmazonS3/latest/userguide/BucketRestrictions.html), buckets have no limit on the number of objects they can contain.  If your objects are to be uploaded on a regular schedule, you could use [bucket lifecycles](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lifecycle-mgmt.html) to have older objects be deleted.  Otherwise, you're right that you'd have to use Lambda or some other service to automate these deletions.  Please let me know if you have any other questions.  

okay thanks for the help  , however 
let's say i did lifecycle that will delete the objects every 3 days while i upload them everyday from ghe server but 

in one day something wrong happened and the server didn't send backup to the bucket 

if i didn't noticed this the objects will be deleted after 3 days and ill not have any backup in the bucket

am i right ? and how to avoid this

**Reply by RyanFitzSimmonsAK** [reply]
You are correct that the bucket would continue to delete objects as they expire.  If you are concerned about upload failures, you could use [AWS Backup](https://docs.aws.amazon.com/aws-backup/latest/devguide/s3-backups.html) to backup your S3 bucket, and be able to restore to a point in time prior to your upload failures.  Alternatively, you could try a [CloudWatch alarm](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ConsoleAlarms.html) based on the number of objects in the bucket.
